### PR TITLE
4549 Insert staff payments into incorp app correction and make this a common component

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
Issue #: /bcgov/entity#4549

Description of changes:
 - StaffPayment component 
 - Setup testing 
 - Storybook config

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: src/components/DetailComment/DetailComment.stories.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+  argTypes: {
+  }
 }
 
-export const Default = () => ({
+const Template = (args, { argTypes }) => ({
+  props: Object.keys(argTypes),
   components: { DetailComment },
-  template: '<detail-comment />'
+  template: '<detail-comment v-bind="$props" />'
```
Comment (severinbeauvais) [inline src/components/DetailComment/DetailComment.stories.ts:13]: This is cool but I'd like to see a comment that `$props` comes from `args` below.

File: src/components/StaffPayment/StaffPayment.stories.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const Template = (args, { argTypes }) => ({
+  props: Object.keys(argTypes),
+  components: { StaffPayment },
+  template: '<staff-payment v-bind="$props" />'
+})
+
+export const Default = Template.bind({})
+Default.args = {
+  staffPaymentData: {
```
Comment (severinbeauvais) [inline src/components/StaffPayment/StaffPayment.stories.ts:19]: Would it be good to specify the interface here? Eg,
`Default.args: StaffPaymentIF = { ... }`

File: src/components/StaffPayment/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    "name": "@bcrs-shared-components/staff-payment",
+    "version": "1.0.3",
+    "publishConfig": {
+        "access": "public"
+    },
+    "dependencies": {
+        "@bcrs-shared-components/enums": "^1.0.0",
+        "@bcrs-shared-components/interfaces": "^1.0.2",
+        "vue-property-decorator": "^8.4.0"
+    }
```
Comment (severinbeauvais) [inline src/components/StaffPayment/package.json:11]: Why is Vuetify not needed here?

File: src/shims-tsx.d.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  namespace JSX {
+    // tslint:disable no-empty-interface
+    interface Element extends VNode {}
+    // tslint:disable no-empty-interface
+    interface ElementClass extends Vue {}
+    interface IntrinsicElements {
+      [elem: string]: any
+    }
+  }
+}
```
Comment (severinbeauvais) [inline src/shims-tsx.d.ts:13]: Is this file needed? If not, please remove it.

Also for src/shims-vue.d.ts.

Also for src/utils/sleep.ts.

**Review by severinbeauvais — APPROVED** [review]
Awesome updates!

Nice cleanup of files to ignore! 👍 

Do you have steps to run the Jest tests?

I see you are committing a copy of the fonts. Is there an alternative? I know they're just used for Storybook here, but still it's another repo to update if they change. (This is not a big deal though.)

Looks good to me!

File: src/shims-tsx.d.ts (hunk 4 above)
Comment (vysakh-menon-aot) [reply src/shims-tsx.d.ts:13]: Yes. shims-vue to avoid vs code lint error for .vue file and sleep is been used in unit test

File: src/components/StaffPayment/package.json (hunk 3 above)
Comment (vysakh-menon-aot) [reply src/components/StaffPayment/package.json:11]: then we have to mention all other dependencies including vue. do you think its needed?

File: src/components/StaffPayment/StaffPayment.stories.ts (hunk 2 above)
Comment (vysakh-menon-aot) [reply src/components/StaffPayment/StaffPayment.stories.ts:19]: Will do the change

Comment (severinbeauvais) [reply src/components/StaffPayment/StaffPayment.stories.ts:19]: Cool.
It probably doesn't help much for Storybook, but it makes it explicit if a developer copies this code.

File: src/components/StaffPayment/package.json (hunk 3 above)
Comment (severinbeauvais) [reply src/components/StaffPayment/package.json:11]: Upon thinking about this, I believe you are right: this isn't needed.

Currently, in this project, Vue/Vuetify is declared by Storybook (in `preview.js`). This is similar to the app, where `main.ts` declares Vue/Vuetify. In both cases, the component doesn't need to know anything about this.

Therefore, here, I agree that the component doesn't need to import Vue or Vuetify -- it should just expect Storybook/app to provide it.
